libs/chonky/src/lib.rs

```diff
@@ -1,26 +1,134 @@
 #![doc = include_str!("../README.md")]
 
-/// Adds two numbers together
-///
-/// # Example
-///
-/// ```rust
-/// use chonky::add;
-///
-/// assert_eq!(add(1, 3), 4);
-/// ```
-#[must_use]
-pub const fn add(left: u64, right: u64) -> u64 {
-    left + right
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum ChonkyError {
+    #[error("parsing error in pdf")]
+    ReadPdf,
+    #[error("pdfium error")]
+    Pdfium,
+    #[error("write error to system")]
+    Write,
+    #[error("Issues with CLI input")]
+    Arguments,
+}
+
+pub mod pdf_segmentation {
```

Comment:
It's good practice to group functionality, thus a module makes sense. However, defining inline modules is quite rare (In this case the size of the module is quite small so it doesn't matter). In most cases you just define the module as in `pub mod pdf_segmentation;` and add a `pdf_segmentation.rs` with the content. (We don't need to do this now, maybe in the future it makes sense, but I guess as the crate grows we will probably shake the module tree quite a few times.

Reply:
Ok I was planning on doing this once the codebase expands slightly more (as in more modules arise), but it would be better practice to start implementing the module file system now instead of later.